storage/src/vespa/storage/storageserver/statemanager.cpp

```diff
@@ -520,21 +521,25 @@ StateManager::run(framework::ThreadHandle& thread)
 {
     while (true) {
         thread.registerTick();
-        std::unique_lock guard(_threadLock);
-            // Take lock before doing stuff, to be sure we don't wait after
-            // destructor have grabbed lock to stop() us.
         if (thread.interrupted()) {
             break;
         }
         tick();
-        _threadCond.wait_for(guard, 1000ms);
+        std::unique_lock guard(_threadLock);
```

Comment:
Is it safe to move `tick()` outside of `_threadLock`? Or does it just touch stuff related to `_stateLock`?

Reply:
`_threadLock` was only used to be able to sleep on a condition variable, it did not guard anything.